snow/consensus/snowball/binary_snowflake.go

```diff
@@ -50,26 +59,34 @@ func (sf *binarySnowflake) RecordPoll(count, choice int) {
 		return
 	}
 
-	if count < sf.alphaConfidence {
-		sf.confidence = 0
+	// If I need to change my preference, record the new preference
+	// and reset all my confidence counters.
+	if choice != sf.Preference() {
 		sf.binarySlush.RecordSuccessfulPoll(choice)
```

Comment:
Same comment as this other place... just feels odd only to call `RecordSuccessfulPoll` if it is for a different color...